src/bin/cargo-deadlinks.rs

```diff
@@ -85,28 +91,39 @@ fn main() {
 /// from the package name found there.
 ///
 /// All *.html files under the root directory will be checked.
-fn determine_dir() -> Vec<PathBuf> {
+fn determine_dir(no_build: bool) -> Vec<PathBuf> {
     let manifest = MetadataCommand::new()
         .no_deps()
         .exec()
         .unwrap_or_else(|err| {
             println!("error: {}", err);
             println!("help: if this is not a cargo directory, use `--dir`");
-            ::std::process::exit(1);
+            process::exit(1);
         });
     let doc = manifest.target_directory.join("doc");
 
     // originally written with this impressively bad jq query:
     // `.packages[] |select(.source == null) | .targets[] | select(.kind[] | contains("test") | not) | .name`
-    manifest
+    let paths = manifest
         .packages
         .into_iter()
         .filter(|package| package.source.is_none())
         .map(|package| package.targets)
         .flatten()
         .filter(has_docs)
         .map(|target| doc.join(target.name.replace('-', "_")))
-        .collect()
+        .collect();
```

Comment:
Rather than coming up with the paths yourself manually, you can get them from `cargo` during the build:

```console
> cargo doc --no-deps --message-format json 2>/dev/null | jq -r 'select(.reason == "compiler-artifact") | .filenames[] | select(endswith("/index.html")) | rtrimstr("/index.html")'
/home/nemo157/.cargo/shared-target/doc/async_compression
```

Which also deals with target specific directories for you:

```console
> cargo doc --target x86_64-unknown-linux-gnu --no-deps --message-format json 2>/dev/null | jq -r 'select(.reason == "compiler-artifact") | .filenames[] | select(endswith("/index.html")) | rtrimstr("/index.html")'
/home/nemo157/.cargo/shared-target/x86_64-unknown-linux-gnu/doc/async_compression
```

Reply:
Great suggestion, thanks!